packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -168,3 +171,132 @@ export namespace ProblemDecorator {
         (left.data.severity || Number.MAX_SAFE_INTEGER) - (right.data.severity || Number.MAX_SAFE_INTEGER);
 
 }
+
+@injectable()
+export class ProblemTabBarDecorator implements TabBarDecorator {
+
+    readonly id = 'theia-problem-tabbar-decorator';
+
+    protected readonly emitter: Emitter<(titles: Title<Widget>[]) => Map<string, TabBarDecoration.Data>>;
+
+    constructor(
```

Comment:
please review coding guidelines and write clean new code, don't copy & paste old code from other places

in this case use property injections

Reply:
Sure. Just fixed it ;).